modules/serving-transport-encryption-choice-of-a-certificate-issuer.adoc

```diff
@@ -0,0 +1,49 @@
+:_mod-docs-content-type: CONCEPT
+[id="serving-transport-encryption-choice-of-a-certificate-issuer_{context}"]
+= Choice of a certificate issuer
+
+[NOTE]
+====
+Issuers refer to cert-manager `Issuers` and `ClusterIssuers`. They represent certificate authorities (CAs) that can generate signed certificates by honoring certificate signing requests.
+====
+
+To identify your certificate issuer, use `cert-manager` documentation, which has examples for the following:
+
+* A custom CA stored in a K8s secret
+* HTTP-01 challenges, e.g. Let’s encrypt
+* DNS-01 challenges
+* Self-signed issuers
+
+[NOTE]
+====
+Not all issuer types work for each Knative Serving encryption feature.
+====
+
+Depending on the encryption part you use, {ServerlessProductName} requires your certificate issuer to be able to sign certain certificates.
+
+* For cluster-local encryption, the issuer must be able to sign certificates for cluster-local domains of these types:
++
+--
+** `myapp.<namespace>`
+** `myapp.<namespace>.svc`
+** `myapp.<namespace>.svc.cluster.local`
+--
++
+As the CA usually is not within the cluster, verification using the ACME protocol (DNS01/HTTP01) is not possible. You can use an issuer that allows creating these certificates (e.g. cert-manager’s CA issuer).
+
+* For system-internal encryption, the issuer must be able to sign certificates with the following Subject Alternative Names (SAN):
++
+--
+** `kn-routing`
+** `kn-user-<namespace>`, where `<namespace>` is each namespace where Knative Services are created
+** `data-plane.knative.dev`
+--
++
+These SANs are needed for Knative to verify connections between the internal components. As this is also not possible using the ACME protocol (DNS01/HTTP01), you need to configure an issuer that allows creating these certificates, for example, `cert-manager` CA issuer.
+
+[id="additional-resources_serving-transport-encryption-choice-of-a-certificate-issuer"]
+[role="_additional-resources"]
+== Additional resources
+
+* link:https://cert-manager.io/docs/concepts/issuer/[cert-manager documentation on issuers]
+* link:https://cert-manager.io/docs/configuration/issuers/[List of cert-manager integrations]
```

Comment:
Are there downstream docs available for this considering https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift ?

Reply:
None of the cert-manager.io links that are in this PR have a direct counterpart in the downstream docs.